src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -101,20 +101,23 @@ private UserDto mapUserToDto(User user) {
         return convertToDto(parent.getChildren(), ChildDto.class);
     }
 
-    public void addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
+    public List<ChildDto> addChildren(User parent, List<ChildDto> childrenDtoToAdd) {
         final List<Child> childrenToSave = new ArrayList<>(convertAllTo(
                 childrenDtoToAdd, Child.class, ArrayList::new));
         addNewChildren(parent, childrenToSave);
         userRepository.save(parent);
+        return findDuplicatedChildAndConvertToDto(parent.getChildren(), childrenToSave);
     }
 
-    public void updateChildren(User parent, List<ChildDto> childrenDtoToUpdate) {
+    public List<ChildDto> updateChildren(User parent, List<ChildDto> childrenDtoToUpdate) {
         parent.getChildren().forEach(pChild -> childrenDtoToUpdate.forEach(uChild -> {
             if (pChild.getId() == uChild.getId()) {
                 BeanUtils.copyProperties(uChild, pChild);
             }
         }));
         userRepository.saveAndFlush(parent);
+        return findDuplicatedChildAndConvertToDto(parent.getChildren(),
+            new ArrayList<>(convertAllTo(childrenDtoToUpdate, Child.class, ArrayList::new)));
```

Comment:
Wow, you method name indicated that it makes convertation but you do it here one more time... Why?

Reply:
First I convert childrenDtoToUpdate to Child.class, after that I find duplicates (because I can't do that with ChilDto) and at last I convert finded childs to dtos for response